File: matching-service/internal/domain/entity/match.go
package entity

import "time"

type MatchStatus string

const (
	MatchStatusActive    MatchStatus = "active"
	MatchStatusExpired   MatchStatus = "expired"
	MatchStatusBlocked   MatchStatus = "blocked"
)

type Match struct {
	ID                 int64
	User1ID            int64
	User2ID            int64
	CreatedAt          time.Time
	Status             MatchStatus
	ConversationStarted bool
	LastInteractionAt  time.Time
}

func NewMatch(user1ID, user2ID int64) *Match {
	now := time.Now()
	return &Match{
		User1ID:            user1ID,
		User2ID:            user2ID,
		CreatedAt:          now,
		Status:             MatchStatusActive,
		ConversationStarted: false,
		LastInteractionAt:  now,
	}
}

func (m *Match) StartConversation() {
	m.ConversationStarted = true
	m.LastInteractionAt = time.Now()
}

File: gateway-service/app/domain/match.py
from dataclasses import dataclass


@dataclass
class Match:
    id: int
    user1_id: int
    user2_id: int
    status: str
    conversation_started: bool

    @classmethod
    def from_dict(cls, data: dict) -> "Match":
        m = data.get("match", data)
        return cls(
            id=int(m["id"]),
            user1_id=int(m["user1Id"]),
            user2_id=int(m["user2Id"]),
            status=m.get("status", "active"),
            conversation_started=m.get("conversationStarted", False),
        )
